drivers/flash/flash_stm32_ospi.c

```diff
@@ -0,0 +1,1498 @@
+	}
+
+	/* Enable write operations */
+	if (stm32_ospi_write_enable(&dev_data->hospi,
+		dev_cfg->data_mode, dev_cfg->data_rate) != 0) {
+		LOG_DBG("Erase failed : write enable");
+		return -ENOTSUP;
+	}
+
+	cmd_erase.InstructionMode    = (dev_cfg->data_mode == OSPI_SPI_MODE)
+					? HAL_OSPI_INSTRUCTION_1_LINE
+					: HAL_OSPI_INSTRUCTION_8_LINES;
+	cmd_erase.InstructionDtrMode = (dev_cfg->data_rate == OSPI_DTR_TRANSFER)
+					? HAL_OSPI_INSTRUCTION_DTR_ENABLE
+					: HAL_OSPI_INSTRUCTION_DTR_DISABLE;
+	cmd_erase.InstructionSize    = (dev_cfg->data_mode == OSPI_SPI_MODE)
+					? HAL_OSPI_INSTRUCTION_8_BITS
+					: HAL_OSPI_INSTRUCTION_16_BITS;
+
+	while ((size > 0) && (ret == 0)) {
+		if (size == dev_cfg->flash_size) {
+			/* chip erase */
+			LOG_INF("Chip Erase");
+			cmd_erase.Instruction = (dev_cfg->data_mode == OSPI_SPI_MODE)
+					? SPI_NOR_CMD_BULKE
+					: SPI_NOR_OCMD_BULKE;
+			cmd_erase.AddressMode = HAL_OSPI_ADDRESS_NONE;
+			/* full chip erase command */
+			ospi_send_cmd(dev, &cmd_erase);
+
+			size -= dev_cfg->flash_size;
+		} else {
+			/* sector erase */
+			LOG_INF("Sector Erase");
+			cmd_erase.Address = addr;
+			const struct jesd216_erase_type *erase_types =
+							dev_data->erase_types;
+			const struct jesd216_erase_type *bet = NULL;
+
+			for (uint8_t ei = 0;
+				ei < JESD216_NUM_ERASE_TYPES; ++ei) {
+				const struct jesd216_erase_type *etp =
+							&erase_types[ei];
+
+				if ((etp->exp != 0)
+				    && SPI_NOR_IS_ALIGNED(addr, etp->exp)
+				    && SPI_NOR_IS_ALIGNED(size, etp->exp)
+				    && ((bet == NULL)
+					|| (etp->exp > bet->exp))) {
+					bet = etp;
+					cmd_erase.Instruction = bet->cmd;
+				} else {
+					/* use the default sector erase cmd */
+					cmd_erase.Instruction =
+						(dev_cfg->data_mode == OSPI_SPI_MODE)
+						? SPI_NOR_CMD_SE /* erase sector size 3 bytes */
+						: SPI_NOR_OCMD_SE;
+					cmd_erase.AddressMode =
+						(dev_cfg->data_mode == OSPI_SPI_MODE)
+						? HAL_OSPI_ADDRESS_1_LINE
+						: HAL_OSPI_ADDRESS_8_LINES;
+					cmd_erase.AddressDtrMode =
+						(dev_cfg->data_rate == OSPI_DTR_TRANSFER)
+						? HAL_OSPI_ADDRESS_DTR_ENABLE
+						: HAL_OSPI_ADDRESS_DTR_DISABLE;
+					cmd_erase.AddressSize = HAL_OSPI_ADDRESS_32_BITS;
+					cmd_erase.Address = addr;
+				}
+			}
+			ospi_send_cmd(dev, &cmd_erase);
+			if (bet != NULL) {
+				addr += BIT(bet->exp);
+				size -= BIT(bet->exp);
+			} else {
+				addr += SPI_NOR_SECTOR_SIZE;
+				size -= SPI_NOR_SECTOR_SIZE;
+			}
+		ret = stm32_ospi_mem_ready(&dev_data->hospi,
+						 dev_cfg->data_mode, dev_cfg->data_rate);
+		}
+	}
+
+	ospi_unlock_thread(dev);
+
+	return ret;
+}
+
+/* Function to read the flash with possible OSPI/SPI and STR/DTR */
+static int flash_stm32_ospi_read(const struct device *dev, off_t addr,
+				 void *data, size_t size)
+{
+	const struct flash_stm32_ospi_config *dev_cfg = DEV_CFG(dev);
+	int ret;
+
+	if (!ospi_address_is_valid(dev, addr, size)) {
+		LOG_DBG("Error: address or size exceeds expected values: "
```

Comment:
```suggestion
		LOG_ERR("Error: address or size exceeds expected values: "
```
That looks important to be seen.